// home-engine/objects/lamp.js

// Constructor
function Lamp(label, identifier) {
  // always initialize all instance properties
  this.label = label;
  this.identifier = identifier;
  this.status = 'turned_off';
  this.watts_used = 0;
  
  this.watt_usage_per_second = 60;
  this.last_turned_on_date = null;
  this.last_turned_off_date = null;
}
// class methods
Lamp.prototype.turnOn = function() {
	this.status = 'turned_on';
	this.last_turned_on_date = new Date().getTime();
	engine.sendMessage(this.identifier, ';TRUE');
	return '';
}
Lamp.prototype.turnOff = function() {
	this.status = 'turned_off';
	this.last_turned_off_date = new Date().getTime();
	
	if(this.last_turned_on_date != null){
		this.watt_usage_per_second = this.watt_usage_per_second + 
					( (this.watt_usage_per_second) * ( (this.last_turned_off_date - this.last_turned_on_date)/1000 ) );
	}
	this.last_turned_on_date = null;
	engine.sendMessage(this.identifier, ';FALSE');
	return '';
}
Lamp.prototype.getStatus = function() {
	return this.status;
}
Lamp.prototype.getWatts = function(){
	return this.watts_used;
}

Lamp.prototype.parseMessage = function(message){
	
};

// export the class
module.exports = Lamp;
