// home-engine/engine.js
var VirtualHome = require('./connections/virtual_home.js');
var Objects = require('./objects/index.js');

var LampSaver = require('./energy/lamp_saver.js');
var Alarm = require('./security/alarm.js');


exports.init = function(){
	this.objects = new Objects();
	global.$ = this.objects.init();
	
	this.connectionTypes = new Object();
	
	this.virtual_home = new VirtualHome();
	
	this.connectionTypes['bedroom'] = this.virtual_home;
	this.connectionTypes['living'] = this.virtual_home;
	this.connectionTypes['bathroom'] = this.virtual_home;
	this.connectionTypes['kitchen'] = this.virtual_home;
	this.connectionTypes['hol'] = this.virtual_home;
	
	
	this.lamp_saver = new LampSaver();
	this.alarm = new Alarm();
	
	
	$('home').lock();
	
};

exports.sendMessage = function(identifier, message){
	//TODO: connection type
	this.connectionTypes[identifier].sendMessage(identifier, message);
};

exports.process = function(name, message){
	answer = "";
	if ( message.command != undefined ){
		answer = this.process_command( message.command ); 
	}
	
	//var data = { 'message' : answer, pseudo : name };
	//soket.broadcast.emit('message', data);
};

exports.process_command = function( command ){
	answer = '';
	switch ( command.name ){
		case 'turnOn':
			$(command.value.toLowerCase()).turnOn();
			break;
		case 'turnOff':
			$(command.value.toLowerCase() ).turnOff();
			break;
		default:
			break;
	}
	return '';
};
